signac/__main__.py

```diff
@@ -1120,6 +1138,44 @@ def main():
         help="Sort the state point keys for output.")
     parser_statepoint.set_defaults(func=main_statepoint)
 
+    parser_diff = subparsers.add_parser(
+        'diff',
+        description="Find the difference among job state points.")
+    parser_diff.add_argument(
+        'job_id',
+        nargs='*',
+        type=str,
+        help="One or more job ids. The jobs must be initialized.")
+    parser_diff.add_argument(
+        '-p', '--pretty',
+        type=int,
+        nargs='?',
+        const=3,
+        help="Print state point in pretty format. "
+             "An optional argument to this flag specifies the maximal "
+             "depth a state point is printed.")
+    parser_diff.add_argument(
+        '-i', '--indent',
+        type=int,
+        nargs='?',
+        const='2',
+        help="Specify the indentation of the JSON formatted state point.")
+    parser_diff.add_argument(
+        '-f', '--filter',
+        type=str,
+        nargs='+',
+        help="Limit the diff to jobs matching this state point filter.")
+    parser_diff.add_argument(
+        '-d', '--doc-filter',
+        type=str,
+        nargs='+',
+        help="Show documents of jobs matching this document filter.")
+    parser_diff.add_argument(
+        '-s', '--sort',
+        action='store_true',
+        help="Sort the state point keys for output.")
```

Comment:
Is the sorting actually implemented? If so, I'd remove this option for now.

Reply:
that was my original intention, but I didn't get around to it and I'll take it out for now.